dist/Storable/t/blessed.t

```diff
@@ -414,3 +414,54 @@ is(ref $t, 'STRESS_THE_STACK');
 
     unlink("store$$");
 }
+
+{
+    # trying to freeze a glob via STORABLE_freeze
+    {
+        package GlobHookedBase;
+
+        sub STORABLE_freeze {
+            return \1;
+        }
+
+        package GlobHooked;
+        our @ISA = "GlobHookedBase";
+    }
+    use Symbol ();
+    my $glob = bless Symbol::gensym(), "GlobHooked";
+    eval {
+        my $data = freeze($glob);
+    };
+    my $msg = $@;
+    like($msg, qr/Unexpected object type \(GLOB\) of class 'GlobHooked' in store_hook\(\) calling GlobHookedBase::STORABLE_freeze/,
+         "check we get the verbose message");
+}
+
+SKIP:
+{
+    $] < 5.012
+      and skip "Can't assign regexps directly before 5.12", 4;
+    my $hook_called;
+    # store regexp via hook
+    {
+        package RegexpHooked;
+        sub STORABLE_freeze {
+            ++$hook_called;
+            "$_[0]";
+        }
+        sub STORABLE_thaw {
+            my ($obj, $cloning, $serialized) = @_;
+            ++$hook_called;
+            $$obj = ${ qr/$serialized/ };
```

Comment:
Wow. I am shocked that works. IN C sure, but in perl, im pretty surprised.  I guess it is safe because storable isnt backported?

Reply:
Backported?  I wrote that a while ago, and from the condition it worked in 5.12, and it tested ok in 5.24 in CI.

There's code in sv_setsv() to handle REGEXP assignment, at least in blead.

It looks like deliberate support was added in f0826785082.

Reply:
Oh wait. It doesn't actually work properly. `$obj` is not blessed. but `qr/.../` is. I guess that is expected?

Reply:
Thanks for the pointer. Interesting what people do with functions you create. (I created reg_temp_copy) I didnt expect this use case at all. :-) Makes perfect sense tho.